core/src/Cabana_CommunicationPlan.hpp

```diff
@@ -250,19 +251,23 @@ class CommunicationPlan
             num_export_host( _num_export.data(), num_n );
         auto export_counts = Kokkos::create_mirror_view_and_copy(
                 memory_space(), num_export_host );
+        auto export_counts_sv =
+            Kokkos::Experimental::create_scatter_view( export_counts );
```

Comment:
Are we worried about always having to create the views? Is it worth considering providing a fixed size allocation for us by experts with consistent/regular data patterns/sends?

Reply:
I guess then we have to be more careful about resetting the data too

Reply:
I guess I'm not sure what you mean here.

Reply:
Sorry, I should have been clearer.  This makes a `ScatterView` per function invocation, right? And then throws it away at the end? 

For regular/structured apps, does this imply memory allocation/deallocation of the same size every timestep? Or is this somehow only ever done once and I'm missing a detail?

Reply:
I believe it would imply allocation/deallocation of the replicated views if replication is the scatter method of choice. Because we don't have information about the slice the user wants to scatter a priori I think it would be difficult to save the state of a scatter view somewhere unless we developed some type of container approach where the user would register the slices they are scattering.

Reply:
Yea, that's what I'm thinking. A second advance API where they provide more information. Something for down the road perhaps.